src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpNamingStyleDiagnosticAnalyzer.cs

```diff
@@ -8,9 +10,27 @@ namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.NamingStyles
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     internal sealed class CSharpNamingStyleDiagnosticAnalyzer : NamingStyleDiagnosticAnalyzerBase
     {
-        public CSharpNamingStyleDiagnosticAnalyzer()
+        protected sealed override void OnCompilationStartAction(
+            CompilationStartAnalysisContext context,
+            ConcurrentDictionary<Guid, ConcurrentDictionary<string, string>> idToCachedResult)
         {
+            // HACK: RegisterSymbolAction doesn't work with local functions
+            context.RegisterSyntaxNodeAction(c => SyntaxNodeAction(c, idToCachedResult), SyntaxKind.LocalFunctionStatement);
+        }
+
+        private void SyntaxNodeAction(
+            SyntaxNodeAnalysisContext context,
+            ConcurrentDictionary<Guid, ConcurrentDictionary<string, string>> idToCachedResult)
+        {
+            var symbolContext = new SymbolAnalysisContext(
```

Comment:
I'm pretty surprised this even had a public constructor you could use in this manner.  @mavasani expected?

Reply:
me too

Reply:
resolved in 3a806e7f175d0985cd9170a78e4d9ba2bc2bf7c3